### PR TITLE
Feature/add file upload

### PR DESCRIPTION
**Pull request by laggery**

### CODE REVIEW

File: src/fileupload/file-upload.controller.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+@Controller('file')
+export class FileUploadController {
+
+  constructor(private fileUploadService: FileUploadService) {
+  }
+
+  @UseGuards(JwtAuthGuard)
+  @Post('upload')
+  @UseInterceptors(FileInterceptor('file'))
+  async uploadFile(@Request() req, @UploadedFile() file: Express.Multer.File, @Query() query) {
```
Comment (fabiangotzen) [inline src/fileupload/file-upload.controller.ts:27]: Can we remove the query parameter here?

File: src/fileupload/file-upload.service.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        Bucket: this.bucket,
+        Key: `${this.env}/${userId}/${file.originalname}`,
+        Body: file.buffer,
+        UploadId: userId.toString(),
+        ACL: 'private'
+      };
+
+      const command = new PutObjectCommand(params);
+
+      const res = await this.s3.send(command);
```
Comment (fabiangotzen) [inline src/fileupload/file-upload.service.ts:45]: const res = 
can be removed

File: src/fileupload/file-upload.service.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    try {
+      const params = {
+        Bucket: this.bucket,
+        CopySource: `${this.env}/${userId}/${copyFile.sourceFileName}`,
+        Key: `${this.env}/${userId}/${copyFile.destinationFileName}`,
+        ACL: 'private'
+      };
+
+      const command = new CopyObjectCommand(params);
+      const res = await this.s3.send(command);
```
Comment (fabiangotzen) [inline src/fileupload/file-upload.service.ts:61]: const res = 
Can we remove this?

File: src/fileupload/file-upload.service.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+    const params = {
+      Bucket: this.bucket,
+      Key: `${this.env}/${userId}/${filename}`
+    };
+
+    const command = new GetObjectCommand(params);
+
+    try {
+      const res = await this.s3.send(command);
```
Comment (fabiangotzen) [inline src/fileupload/file-upload.service.ts:85]: const res = 
Can we remove this?

File: src/fileupload/file-upload.service.ts [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  async deleteFile(userId: number, filename: string) {
+    const params = {
+      Bucket: this.bucket,
+      Key: `${this.env}/${userId}/${filename}`,
+    };
+
+    const command = new DeleteObjectCommand(params);
+
+    try {
+      const res = await this.s3.send(command);
```
Comment (fabiangotzen) [inline src/fileupload/file-upload.service.ts:101]: const res = 
Can we remove this?

File: src/fileupload/file-upload.controller.ts (hunk 1 above)
Comment (laggery) [reply src/fileupload/file-upload.controller.ts:27]: Done

File: src/fileupload/file-upload.service.ts (hunk 2 above)
Comment (laggery) [reply src/fileupload/file-upload.service.ts:45]: Done

File: src/fileupload/file-upload.service.ts (hunk 3 above)
Comment (laggery) [reply src/fileupload/file-upload.service.ts:61]: Done

File: src/fileupload/file-upload.service.ts (hunk 5 above)
Comment (laggery) [reply src/fileupload/file-upload.service.ts:101]: Done

File: src/fileupload/file-upload.service.ts (hunk 4 above)
Comment (laggery) [reply src/fileupload/file-upload.service.ts:85]: I use the res on line 86. so, I can't delete this variable.

**Pull request merged**